### DISCUSSION TITLE
Switching from Xiaomi Mio to Roborock integration

### DISCUSSION THREAD

**Discussion by PskNorz** [PiotrMachowski/lovelace-xiaomi-vacuum-map-card#720] [Q&A]
Hello

I have a couple of questions, because i want to switch from Xiaomi mio integration to the roborock because i having problems lately with my roborock s5 max.

The card will work as it works now with xiaomi mio? I know i need to change some entity names but i will have all the fuctionality that i have now?

What about the map? I will need the map extractor? And i am asking that cause i saw that roborock has its own map.

Thanks

**Comment by PiotrMachowski** [comment]
At this moment Map Card works with [custom Roborock integration](https://github.com/humbertogontijo/homeassistant-roborock)

**Reply by PskNorz** [reply]
So to have full fuctionality like i have now with xiaomi mio, i need to use the integrations from the link?

**Reply by PiotrMachowski** [reply]
Yes, unfortunately built-in integration doesn't fully work with this card at this moment

**Comment by PskNorz** [comment]
Ok thank you i will give it a try
